internal/server/server.go

```diff
@@ -35,21 +35,22 @@ func requestHandler(ctx *fasthttp.RequestCtx) {
 	req.SetBody(ctx.Request.Body())
 
 	host := parseHost(ctx)
-	route, err := store.GetRouteByDomain(host)
+	route, err := store.GetRouteByDomainKey(host)
 	if err != nil {
 		ctx.Response.SetStatusCode(404)
 		ctx.Response.SetBodyString(fmt.Sprintf("Domain %s not found", ctx.Host()))
 	} else { // the route exists so we should have a deployment attached to the service
-		origin := fmt.Sprintf("%s:%s", route.Service, route.Port)
+		origin := fmt.Sprintf("%s.%s:%s", route.Service, route.Namespace, route.Port)
 		req.SetHost(origin)
 
 		store.UpdateLastUse(host)                       // see how we can avoid doing that every time
 		if err := httpClient.Do(req, res); err != nil { // First try
-			log.Debug().Msg("Error forwarding the request - Scaling up the deployment")
+			log.Debug().Msg("Error forwarding the request - Try scaling up the deployment")
 			// the deployment is scaled down, let's scale it up
-			if err := kubernetes.ScaleUp(route.Label, route.Namespace); err != nil {
+			if err := upscaler.ScaleUpDeployment(route.Deployment, route.Namespace); err != nil {
 				forwardError(ctx, err)
 			} else { // Second try with the deployment scaled up
+				store.UpdateLastUse(host) // TODO remove that - we are updating again last use because of the timeout
 				if err := httpClient.Do(req, res); err != nil {
```

Comment:
would this call ever be successful after calling scale up deployment?

Reply:
Or is this setup with a retry loop? It would be good to throw this in a function or something if its doing something special, otherwise reading it it looks like its just gonna error and the client has to try again manually versus retrying until a timeout. It just looks like magic or a non advance solution compared to knative or something that reties. I'm thinking u do have it retrying somewhere but if I could understand how this functionality works while reading this code here, it would be helpful. (comment, functions /w names, etc)

Reply:
There is not retry loop.
This is the retry itself.
The `ScaleUpDeployment` is blocking. And it will wait for N seconds before retrying.
If the pod is not up after N second, it will timeout (the timeout code is not implemented yet though).

Don't compare proxless and knative. 😛
Knative has a queue proxy deployed as a sidecar on everypod and queue all the requests. U can even specify a rate limiter there. So this is definitely a most advanced solution.

Proxless goal is to be much simpler. So u will for sure lose that kind of feature. But who need it? I believe that people like us who wanna use serverless solution does this for cost saving. Performance can come later.

For example, on top of my head, some advantages that proxless will have compared to knative:
- no crd needed
- only 1 dependency -> the proxy itself
- configuration is easy, u just need to add some annotations on ur services
- it will have 2 modes "open" and "restricted". "open" == everything can connect to everything. "restricted" == cross namespace is forbidden. this means it's multi tenant friendly.